client/client.go

```diff
@@ -394,9 +394,34 @@ type RemoteContext struct {
 	// preference.
 	Platforms []string
 
-	// MaxConcurrentDownloads is the max concurrent content downloads for each pull.
+	// MaxConcurrentDownloads restricts the total number of concurrent downloads
+	// across all layers during an image pull operation. This helps control the
+	// overall network bandwidth usage.
 	MaxConcurrentDownloads int
 
+	// MaxConcurrentDownloadOperations limits the number of parallel operations
+	// during an image pull to prevent system resource exhaustion.
+	//
+	// These operations include:
+	// - Downloading a layer
+	// - Unpacking a layer into the snapshotter
+	// - Moving an unpacked layer to its final location
+	MaxConcurrentDownloadOperations int
+
+	// MaxConcurrentDownloadsPerLayer controls parallel downloading of image layers
+	// by splitting them into chunks.
+	//
+	// - Values <= 1: Layers are downloaded using a single connection (default).
+	// - Values > 1: Layers are divided into chunks and downloaded in parallel,
+	//   which can significantly reduce pull times for large layers.
+	MaxConcurrentDownloadsPerLayer int
+
+	// ConcurrentFetchChunksSizeMB sets the maximum size in MB for each chunk
+	// when downloading layers in parallel (requires
+	// MaxConcurrentDownloadsPerLayer > 1). Larger chunks reduce coordination
+	// overhead but use more memory.
+	ConcurrentFetchChunksSizeMB int
```

Comment:
How about passing the size in bytes, instead of MB? We have some `Size` variables and most (all?) of them are in bytes.

https://github.com/search?q=repo%3Acontainerd%2Fcontainerd+size&type=code

Reply:
Good point, updated, after the buffered pipe we need to be more subtle with this setting